### DISCUSSION TITLE
Using Pin in blazor

### DISCUSSION THREAD

**Discussion by pilotshamn** [Mapsui/Mapsui#2144] [Q&A]
Hi Really great that Mapsui for Blazor now works. Mapsui have been fantastic to use  in my Maui App!
But i have a question.
Do Pin exists in Mapsui Blazor and if not is there a replacement. ( Pin.cs seams only to support Maui and Forms)
I have a Maui app that uses the Mapsui class Pin to display Pin objects on a map, now i want to implement this in Blazor.
The Pins are updated every ten seconds, the objects for each pin that is created that are used are Type, Label, Position, Rotation, Anchor, Tag, Scale, Svg/Icon.
At each update I change the Position and Svg (I create the svg dynamically via XmlNode.Attributes) in the Pin and call RefreshGraphics() . This is done to follow ships in "realtime" see the attached image.
Can i solve this in Blazor?
If so any idea on how?
Regards
![AIS](https://github.com/Mapsui/Mapsui/assets/4975613/9f8e0e00-a8d9-41a7-bbaf-af67e3b189a8)

**Comment by inforithmics** [comment]
You can port the relevant code from here for the Pin. 
https://github.com/Mapsui/Mapsui/tree/master/Mapsui.UI.MapView
It is on the Roadmap for the next Mapsui Version.

https://mapsui.com/samples/  All samples are written in Blazor, Maybe you could use a callout as a pin. 
(Info => Callout sample)
or
(Tests => Svg Symbols)

**Comment by pilotshamn** [comment]
Thank you @inforithmics
I used PointFeature like this instead of Pin

```csharp
  pf = new PointFeature(new PointFeature(SphericalMercator.FromLonLat(theShipIn.Lon, theShipIn.Lat).ToMPoint()))
  {
      Styles = new[] { new SymbolStyle {
              SymbolScale = 0.8,
              SymbolOffset = new Offset(theShipIconAnchor),
              BitmapId =BitmapRegistry.Instance.Register(Mapsui.Utilities.SvgHelper.LoadSvgPicture(theBoatSvg)) } }
  };  
  boatfeatures.Add(GetTheFeatureField(pf, theShipIn));
```
and implemented the Callout myself
